packages/eslint-plugin/docs/rules/return-await.mdx

```diff
@@ -260,3 +293,12 @@ async function validNever3() {
 
 </TabItem>
 </Tabs>
+
+### Options Summary
+
+|             Option              | Ordinary Context <br/> (stylistic preference) | Error Handling Context <br/> (catches subtle control flow bugs) |                       Should I use?                        |
+| :-----------------------------: | :-------------------------------------------: | :-------------------------------------------------------------: | :--------------------------------------------------------: |
+|             always              |            `return await promise;`            |                     `return await promise;`                     |                          ✅ Yes!                           |
+|          in-try-catch           |               `return promise;`               |                     `return await promise;`                     |                          ✅ Yes!                           |
+| error-handling-correctness-only |                 don't care 🤷                 |                     `return await promise;`                     | 🟡 Okay to use, but the above options would be preferable. |
+|              never              |               `return promise;`               |                        `return promise;`                        |                           ❌ No!                           |
```

Comment:
[Docs]

> ❌ No!

Well, if the option should _never_ be used... why would we want to include it at all? It's not clear from this table or the <code>## \`never\`</code> section why we have that. Maybe a solution to this would be to mention that `never` is only included for compatibility with the base rule?

Reply:
Documenting a private conversation for visibility:

***

> `never` is only included for compatibility with the base rule

There is no evidence that this is or ever was the case; see https://github.com/typescript-eslint/typescript-eslint/issues/994.

***

Plan to resolve this comment - tentative plan to deprecate 'never' as part of this PR. Getting that process started imminently.

Reply:
filed https://github.com/typescript-eslint/typescript-eslint/issues/9433

Reply:
Incorporated #9433 into this PR